api/v1alpha3/types.go

```diff
@@ -343,7 +343,13 @@ type DataDisk struct {
 
 // ManagedDisk defines the managed disk options for a VM.
 type ManagedDisk struct {
-	StorageAccountType string `json:"storageAccountType"`
+	StorageAccountType string                       `json:"storageAccountType"`
+	DiskEncryptionSet  *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+}
+
+// DiskEncryptionSetParameters defines disk encryption options.
+type DiskEncryptionSetParameters struct {
+	ID string `json:"id,omitempty"`
```

Comment:
Please add a comment to the `DiskEngryptionSetParameters.ID` field. For the naive observer, they might not know the ID is an Azure resource ID.

This isn't from your PR, so no action need for this, but I think folks probably don't know what are valid strings for the `StorageAccountType`. Perhaps, it would be good to provide an enum for the `StorageAccountType`, or provide examples in the comment for the field.

```go
// +kubebuilder:validation:Enum=foo;bar;baz
type StorageAccountType string
```

@CecileRobertMichon wdyt?

Reply:
+1 on the comment for ID format. 

I'll follow up on the StorageAccountType comment.